app/main.py

```diff
@@ -1,3 +1,54 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance) -> Distance:
+        km = other.km if isinstance(other, Distance) else other
+        return Distance(
+            km=self.km + km
+        )
+
+    def __iadd__(self, other: Distance) -> Distance:
+        km = other.km if isinstance(other, Distance) else other
+        self.km += km
+        return self
+
+    def __mul__(self, mul: float) -> Distance:
+        return Distance(
+            km=self.km * mul
+        )
+
+    def __truediv__(self, div: float) -> Distance:
```

Comment:
The `div` parameter in the `__truediv__` method should be annotated as `float` or `int`, as division by an integer is also valid.